libcnb-cargo/src/main.rs

```diff
@@ -0,0 +1,214 @@
+}
+
+fn handle_libcnb_package(matches: &ArgMatches) {
+    let cargo_profile = if matches.is_present("release") {
+        CargoProfile::Release
+    } else {
+        CargoProfile::Dev
+    };
+
+    let target_triple = match matches.value_of("target") {
+        None => {
+            error!("Could not determine target triple!");
+            std::process::exit(1);
+        }
+        Some(target_triple) => target_triple,
+    };
+
+    let current_dir = match std::env::current_dir() {
+        Ok(current_dir) => current_dir,
+        Err(io_error) => {
+            error!("Could not determine current directory: {}", io_error);
+            std::process::exit(1);
+        }
+    };
+
+    info!("Reading buildpack metadata...");
+    let buildpack_data = match read_buildpack_data(&current_dir) {
+        Ok(buildpack_data) => buildpack_data,
+        Err(error) => {
+            error!("Packaging buildpack failed due to an error while reading buildpack metadata!");
+
+            match error {
+                BuildpackDataError::IoError(io_error) => {
+                    error!("IO error while reading buildpack metadata: {}", io_error);
+                    error!("Hint: Verify that a readable file named \"buildpack.toml\" exists at the root of your project.")
+                }
+                BuildpackDataError::DeserializationError(deserialization_error) => {
+                    error!(
+                        "Could not deserialize buildpack metadata: {}",
+                        deserialization_error
+                    );
+                    error!("Hint: Verify that your \"buildpack.toml\" is valid.")
+                }
+            }
+
+            std::process::exit(1);
+        }
+    };
+
+    info!(
+        "Found valid buildpack with id \"{}\" @ {}!",
+        buildpack_data.buildpack_toml.buildpack.id, buildpack_data.buildpack_toml.buildpack.version
+    );
+
+    let output_path = matches
+        .value_of("output-path")
+        .map(PathBuf::from)
+        .or_else(|| {
+            MetadataCommand::new()
+                .manifest_path(&current_dir.join("Cargo.toml"))
+                .exec()
+                .map(|metadata| {
+                    metadata
+                        .target_directory
+                        .join(default_buildpack_tarball_filename(
+                            &buildpack_data.buildpack_toml,
+                            cargo_profile,
+                        ))
+                        .into_std_path_buf()
+                })
+                .ok()
+        });
+
+    let output_path = match output_path {
+        Some(output_path) => output_path,
+        None => {
+            error!("Could not determine output path for tarball!");
+            std::process::exit(1);
+        }
+    };
+
+    let relative_output_path =
+        pathdiff::diff_paths(&output_path, &current_dir).unwrap_or_else(|| output_path.clone());
+
+    info!("Building buildpack binary ({})...", &target_triple);
```

Comment:
We surely could. I included the target_triple there because it's not part of the Cargo build output. The profile will be logged by Cargo itself:

```
INFO - Building buildpack binary (x86_64-unknown-linux-musl)...
    Finished dev [unoptimized + debuginfo] target(s) in 0.06s
```

If you feel strongly, I'm happy to add it to the heading. But I'd not add it to have less noise in the already quite noisy `handle_libcnb_package` function.

Reply:
> The profile will be logged by Cargo itself:

Ah good point. I'm happy to skip this then :-)